src/error.rs

```diff
@@ -58,19 +59,25 @@ impl From<TlsError> for Error {
     }
 }
 
+impl<'a> From<Response<'a>> for Error {
+    fn from(err: Response<'a>) -> Error {
+        Error::Parse(ParseError::Unexpected(format!("{:?}", err)))
+    }
+}
+
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
             Error::Io(ref e) => fmt::Display::fmt(e, f),
             Error::Tls(ref e) => fmt::Display::fmt(e, f),
             Error::TlsHandshake(ref e) => fmt::Display::fmt(e, f),
             Error::Validate(ref e) => fmt::Display::fmt(e, f),
-            Error::BadResponse(ref data) => write!(
-                f,
-                "{}: {}",
-                &String::from(self.description()),
-                &data.join("\n")
-            ),
+            Error::NoResponse(ref data) => {
```

Comment:
These two patterns can be merged `Error::NoResponse(ref data) | Error::BadResponse(ref data)`.